src/librustc/middle/free_region.rs

```diff
@@ -64,7 +64,7 @@ impl FreeRegionMap {
                     // No region bounds here
                 }
                 ty::Predicate::RegionOutlives(ty::Binder(ty::OutlivesPredicate(r_a, r_b))) => {
-                    match (r_a, r_b) {
+                    match (*r_a, *r_b) {
```

Comment:
Nit: I guess we can maybe expect optimizer to handle this, but this seems strictly less efficient than moving the `&` into the patterns...